mongoengine/document.py

```diff
@@ -429,11 +433,18 @@ def _save_create(self, doc, force_insert, write_concern):
         Helper method, should only be used inside save().
         """
         collection = self._get_collection()
-
-        if force_insert:
-            return collection.insert(doc, **write_concern)
-
-        object_id = collection.save(doc, **write_concern)
+        with set_write_concern(collection, write_concern) as wc_collection:
+            if force_insert:
+                return wc_collection.insert_one(doc).inserted_id
+            # insert_one will provoke UniqueError alongside save does not
+            # therefore, it need to catch and call replace_one.
+            if '_id' in doc:
+                raw_object = wc_collection.find_one_and_replace(
+                    {'_id': doc['_id']}, doc)
+                if raw_object:
+                    return doc['_id']
```

Comment:
Could you also confirm that we are testing those 2 cases (with and without `_id`) in the existing test suite? and if not add a test

Reply:
confirmed from code coverage.